File: 1000/2148D/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
 
        while(t-- > 0)
        {
            int n = sc.nextInt();
            long e = 0;
 
            ArrayList<Long> o = new ArrayList<>();
 
            for(int i=0;i<n;i++)
            {
                long x = sc.nextLong();
                if((x&1) == 0)
                    e += x;
                else
                    o.add(x);
            }
 
            if(o.size() == 0)
            {
                System.out.println(0);
                continue;
            }
 
            Collections.sort(o);
 
            int k = o.size(), take = (k+1)/2;
            long ans = e;
 
            for(int i=0;i<take;i++)
                ans += o.get(k-1-i);
 
            System.out.println(ans);
        }
    }
}

File: 1000/2148D/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n;
        cin >> n;

        long long evenSum = 0;
        vector<long long> odd;

        for (int i = 0; i < n; i++) 
        {
            long long x;
            cin >> x;
            if ((x & 1) == 0)
                evenSum += x;
            else
                odd.push_back(x);
        }

        if (odd.empty()) 
        {
            cout << 0 << "\n";
            continue;
        }

        sort(odd.begin(), odd.end());

        int k = odd.size();
        int take = (k + 1) / 2;
        long long ans = evenSum;

        for (int i = 0; i < take; i++)
            ans += odd[k - 1 - i];

        cout << ans << "\n";
    }

    return 0;
}
